src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -86,21 +86,24 @@ public void discoverClusterSlots(Jedis jedis) {
       rediscovering = true;
 
       try {
-        this.slots.clear();
-
         List<Object> slots = jedis.clusterSlots();
 
+        // We should clear slots after getting result about cluster slots, because if we got exception or timeout in
```

Comment:
At first, when we fix JedisSlotBasedConnectionHandler#getConnectionFromSlot, this comment becomes invalid.

And IMO, clearing slots first or later is up to amount of slot changeset. For example, if most of slots are changed, it would be better to try discovering slots again instead of relying on outdated slot information. If only a little of slots are changed, we should take latter. It would be not a big deal cause we don't do random connect later.